packages/venia-concept/src/queries/getProductDetail.graphql

```diff
@@ -18,9 +18,39 @@ query productDetail($urlKey: String, $onServer: Boolean!) {
                 disabled
                 file
             }
+            ... on ConfigurableProduct {
+                configurable_options {
+                    label
+                    attribute_id
+                    attribute_code
+                    position
+                    values{
+                        label
+                        value_index
+                    }
+                }
+                variants {
+                    product {
+                        sku
+                        id
+                        swatch_image
+                        size
```

Comment:
GraphQL is weird. This whole area is a fragment on `ConfigurableProduct`, so you would think that type hint would persist throughout the fragment. Not so!

Our `ConfigurableProduct` type does not specify that its `variants` field is `[ConfigurableProduct]` as well; instead, it's `[ProductInterface]`, since a ConfigurableProduct could have a variant that isn't a ConfigurableProduct. So your fragment identifier needs to also go inside this `product` subselection right here.

Reply:
Alright, I think I got what you meant by this. Changes are on my branch now.